examples/rust-sdk/lp-bot/src/utils.rs

```diff
@@ -0,0 +1,329 @@
+            Ok(balance.ui_amount_string)
+        },
+        MAX_RETRIES,
+        INITIAL_RETRY_DELAY,
+        "fetch token balance",
+    )
+    .await
+}
+
+pub async fn fetch_position(
+    rpc: &RpcClient,
+    position_address: &Pubkey,
+) -> Result<Position, Box<dyn std::error::Error>> {
+    retry_async(
+        || async {
+            let position_account = rpc.get_account(position_address).await?;
+            let position = Position::from_bytes(&position_account.data)?;
+            Ok(position)
+        },
+        MAX_RETRIES,
+        INITIAL_RETRY_DELAY,
+        "fetch position",
+    )
+    .await
+}
+
+pub async fn fetch_whirlpool(
+    rpc: &RpcClient,
+    whirlpool_address: &Pubkey,
+) -> Result<Whirlpool, Box<dyn std::error::Error>> {
+    retry_async(
+        || async {
+            let whirlpool_account = rpc.get_account(whirlpool_address).await?;
+            let whirlpool = Whirlpool::from_bytes(&whirlpool_account.data)?;
+            Ok(whirlpool)
+        },
+        MAX_RETRIES,
+        INITIAL_RETRY_DELAY,
+        "fetch whirlpool",
+    )
+    .await
+}
+
+pub async fn fetch_mint(
+    rpc: &RpcClient,
+    mint_address: &Pubkey,
+) -> Result<Mint, Box<dyn std::error::Error>> {
+    retry_async(
+        || async {
+            let mint_account = rpc.get_account(mint_address).await?;
+            let mint = Mint::unpack(&mint_account.data)?;
+            Ok(mint)
+        },
+        MAX_RETRIES,
+        INITIAL_RETRY_DELAY,
+        "fetch mint",
+    )
+    .await
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+pub enum PriorityFeeTier {
+    None,
+    Low,
+    Medium,
+    High,
+    Turbo,
+}
+
+pub async fn send_transaction(
+    rpc: &RpcClient,
+    wallet: &dyn Signer,
+    instructions: Vec<solana_sdk::instruction::Instruction>,
+    additional_signers: Vec<&dyn Signer>,
+    tier: PriorityFeeTier,
+    max_priority_fee: u64,
+) -> Result<Signature, Box<dyn std::error::Error>> {
+    retry_async(
+        || async {
+            let mut all_instructions = vec![];
+
+            if let Some(priority_fee_instruction) = get_priority_fee_instruction(
+                rpc,
+                &instructions,
+                wallet,
+                &additional_signers,
+                tier,
+                max_priority_fee,
+            )
+            .await?
+            {
+                all_instructions.push(priority_fee_instruction);
+            }
+
+            all_instructions.extend(instructions.clone());
+
+            let recent_blockhash = rpc.get_latest_blockhash().await?;
+            let message = Message::new(&all_instructions, Some(&wallet.pubkey()));
+            let mut all_signers = vec![wallet];
+            all_signers.extend(additional_signers.clone());
+
+            let transaction = Transaction::new(&all_signers, message, recent_blockhash);
+            let signature = rpc.send_and_confirm_transaction(&transaction).await?;
```

Comment:
The best way of sending currently is client retries. I think also confirm_transaction rpc call that send_and_confirm uses is deprecated and removed in solana v2 (which is currently live)